modules/SofaBoundaryCondition/FixedConstraint.inl

```diff
@@ -289,16 +290,25 @@ void FixedConstraint<DataTypes>::projectVelocity(const core::MechanicalParams* m
     const SetIndexArray & indices = this->d_indices.getValue();
     helper::WriteAccessor<DataVecDeriv> res (vData );
 
-    if( d_fixAll.getValue() )    // fix everyting
+    const VecDeriv &freeV = (dynamic_cast<core::behavior::MechanicalState<DataTypes>*>(this->getContext()->getMechanicalState()))->read(core::ConstVecDerivId::freeVelocity())->getValue();
+    size_t freeVsize = freeV.size();
+    if (d_fixAll.getValue() == true)    // fix everyting
     {
         for( unsigned i=0; i<res.size(); i++ )
-            res[i] = Deriv();
+            if (freeVsize > i)
+                res[i] = freeV[i];
+            else
+                res[i] = Deriv();
     }
     else
     {
-        for (SetIndexArray::const_iterator it = indices.begin(); it != indices.end(); ++it)
+        unsigned i = 0;
+        for (SetIndexArray::const_iterator it = indices.begin(); it != indices.end() && i < res.size(); ++it, ++i)
         {
-            res[*it] = Deriv();
+            if (freeVsize > *it)
+                res[*it] = freeV[*it];
```

Comment:
I don't understand why you don't want to set all velocities to Deriv(). Maybe an example can help. tks